core/sky_model.py

```diff
@@ -73,48 +74,59 @@ def dnds_franzen(s, a=None, norm=False):
         return s**-2.5 * out
 
 
-"""===LOGIC CODE==="""
-
-# Commented out due to missing source file
-# def make_eor_model(fch):
-#     """Make EoR SkyModel."""
-#     model_dir = utils.SKYDIR / 'eor'
-#     f = h5py.File(f'{model_dir}/healpix_maps.h5', 'r')
-
-#     nside = int(f['nside'][()])
-#     npix = hp.nside2npix(nside)
-
-#     freqs = f['frequencies_mhz'][:] * 1e6  # units Hz
-#     # nfreqs = len(freqs)
-#     # seed = int(f['realization_random_seed'][()])
-
-#     # HEALPix array -- dimension (nfreqs, npix) -- unit Jy/sr
-#     hmaps = f['healpix_maps_Jy_per_sr'][:, :]
-
-#     # Shift the maps values so there is no negative avalues
-#     hmaps_min = hmaps.min(axis=1)[:, np.newaxis]
-#     hmaps = hmaps - hmaps_min + 0.5 * np.abs(hmaps_min)
-
-#     # Initialize Stoke array
-#     stokes = np.zeros((4, 1, npix)) * units.Jy / units.sr
-#     stokes[0, 0, :] = hmaps[fch] * units.Jy / units.sr
-
-#     # Setup SkyModel params
-#     params = {
-#         'component_type': 'healpix',
-#         'nside': nside,
-#         'hpx_order': 'ring',
-#         'hpx_inds': np.arange(npix),
-#         'spectral_type': 'full',
-#         'freq_array': [freqs[fch]] * units.Hz,
-#         'stokes': stokes
-#     }
-
-#     eor_model = SkyModel(**params)
-#     eor_model.write_skyh5(
-#         f'{model_dir}/fch{fch:04d}.skyh5',
-#         clobber=True
-#     )
+def make_grf_eor_model(model_file: str, channels: list[int]):
+    """Make a GRF EoR SkyModel.
+
+    The model file is here assumed to contain Nfreqs healpix maps. The format of the
+    file is custom, as defined by @zacharymartinot, and the sky map is created by
+    his redshifted_gaussian_field code.
+    """
+    model_dir = utils.SKYDIR / "eor"
+
+    with h5py.File(model_dir / model_file, "r") as fl:
+        nside = int(fl["nside"][()])
+        freqs = fl["frequencies_mhz"][:] * 1e6 << units.Hz  # units Hz
+
+        # HEALPix array -- dimension (nfreqs, npix) -- unit Jy/sr
+        # We must read in the whole thing to set the monopole.
+        hmaps = fl["healpix_maps_Jy_per_sr"][:, :]
+
+    hmaps <<= units.Jy / units.sr
+
+    npix = hp.nside2npix(nside)
+
+    # Shift the maps values so there are no negative values
+    # Note: we move the WHOLE sky up by a set amount so that the minimum value is 1.0.
```

Comment:
"WHOLE sky at ALL frequencies" -- Could be a bit more specific.

Reply:
Good call, fixed.